manifests/init.pp

```diff
@@ -264,6 +265,12 @@
     $apache_https_key = undef
     $apache_https_ca = undef
     $apache_https_chain = undef
+
+    if $foreman_proxy::registration_url {
+      $client_facing_servername = foreman_proxy_content::host_from_url($foreman_proxy::registration_url)
```

Comment:
I was wondering if we should add a failsafe
```suggestion
      $client_facing_servername = foreman_proxy_content::host_from_url($foreman_proxy::registration_url)
      unless $client_facing_servername in $cname {
        fail('...')
      }
```

But then I realized we don't pass `serveraliases` to `pulpcore` at all. In https://github.com/theforeman/puppet-foreman_proxy_content/pull/435/files#r1010674622 I did that and https://github.com/theforeman/puppet-pulpcore/pull/269 has been merged now.

Thoughts on this?

Reply:
I like the idea behind the check